src/main/java/tetris/Controller.java

```diff
@@ -32,24 +32,31 @@
         Platform.runLater(() -> Action.SOFT_DROP.attempt(tetromino, grid));
         Platform.runLater(() -> redraw());
     });
-
     private Settings settings;
+    private TetrominoFactory factory;
 
     /**
      * the Controller class determines the game flow and does the actual event
      * handling.
-     * 
+     *
      * @param ui
-     *            the application in which the game is running
+     * @param settings
+     * @param factory
+     * @param soundManager
+     * @param score
+     * @param scoreBoard
      */
-    public Controller(View ui, Settings settings) {
+    public Controller(View ui, Settings settings, TetrominoFactory factory,
```

Comment:
Actually, I'd recon that utility classes and factory methods in particular more or less serve a role comparable to interfaces.